modules/c++/nitf/tests/test_create_nitf++.cpp

```diff
@@ -0,0 +1,1216 @@
+  "\177\215\232\232\234\222\222\222\211\211\211\202\202\202\202\202\202\206"
+  "\206\206\224\224\224\244\244\244\273\273\273\320\320\320\346\346\346\374"
+  "\374\374\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\374"
+  "\374\374\355\340\340\331\273\273\315\260\260\302\246\246\275\241\241\273"
+  "\237\237\271\236\236\266\233\233\253\222\222\233\204\204yuurrr\214\214\214"
+  "\274\274\274\345\345\345\377\377\377\377\377\377\377\377\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\377\373\373\373\364\364\364\326\267\267"
+  "\314\256\256\303\246\246\276\242\242\273\237\237\272\236\236\272\236\236"
+  "\272\236\236\272\236\236\272\236\236\272\236\236\271\236\236\266\233\233"
+  "\253\222\222\233\204\204yuurrr\214\214\214\276\276\276\347\347\347\372\372"
+  "\372\373\373\373\360\351\351\331\273\273\315\260\260\302\246\246\275\241"
+  "\241\273\237\237\272\236\236\272\236\236\271\236\236\266\233\233\253\222"
+  "\222\233\204\204yuurrr\214\214\214\274\274\274\345\345\345\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\374"
+  "\374\374\361\353\353\331\273\273\315\260\260\302\246\246\275\241\241\273"
+  "\237\237\272\236\236\272\236\236\270\235\235\265\232\232\252\221\221\231"
+  "\202\202uuusss\217\217\217\277\277\277\351\351\351\377\377\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377"
+  "\374\374\374\322\322\332\272\272\304\256\256\270\246\246\260\242\242\253"
+  "\237\237\250\236\236\247\236\236\247\234\234\245\225\225\236\210\210\220"
+  "||\177ooo}}}\244\244\244\322\322\322\367\367\367\377\377\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\377\374\374\374\322\322\332\272\272\304"
+  "\257\257\271\246\246\260\242\242\253\237\237\250\236\236\247\236\236\247"
+  "\235\235\246\232\232\243\221\221\232\203\203\213yyzzzz\225\225\225\303\303"
+  "\303\351\351\351\366\366\366\366\366\366\357\357\357\347\347\347\335\335"
+  "\335\324\324\324\315\315\315\306\306\306\303\303\303\303\303\303\300\300"
+  "\302\245\245\257\243\243\254pp\206oo\205oo\205oo\205oo\205oo\205oo\205\236"
+  "\236\247\234\234\245\256\256\260\251\251\251\235\235\235\223\223\223\215"
+  "\215\215\211\211\211\210\210\210\211\211\211\216\216\216\232\232\232\252"
+  "\252\252\300\300\300\322\322\322\344\344\344\366\366\366\376\376\376\377"
+  "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\376\376\376\371"
+  "\371\371\355\355\355\332\332\332\305\305\305\265\265\265\254\254\254\250"
+  "\250\250\245\245\245\241\241\241\226\226\226\207\207\207{{{\204\204\204\243"
+  "\243\243\320\320\320\366\366\366\377\377\377\377\377\377\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\376\376\376\370\370\370\354\354\354\331"
+  "\331\331\306\306\306\266\266\266\254\254\254\250\250\2
```

Comment:
Can't you compare these directly without a cast?  `nitf_Uint8` will be an `unsigned char`?